integration/webhook_test.go

```diff
@@ -0,0 +1,175 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"github.com/openshift/addon-operator/integration"
+)
+
+const (
+	CATALOG_SOURCE_URL = "quay.io/osd-addons/reference-addon-index@sha256:58cb1c4478a150dc44e6c179d709726516d84db46e4e130a5227d8b76456b5bd"
+	ADDON_NAME         = "reference-addon-test-install-spec"
+)
+
+func TestAddonInstallSpec(t *testing.T) {
+	t.Parallel()
+
```

Comment:
Do you think we should skip this test case when `ENABLE_WEBHOOK` is false?
When we run tests via `make test-integration-short` and the webhook-server is not setup, this test will always fail.

Overall I am not super happy with the UX, when omitting webhooks - but I also don't have many ideas how to improve it :D

Reply:
@thetechnick Made a few changes here

- The execution of `TestAddonInstallSpec` and `TestAddonInstallSpecImmutability` now depend on the `ENABLE_WEBHOOK`
- This means that `make test-integration-short` will always skip the above tests unless they're run with `ENABLE_WEBHOOK=true`
- This also means that `make test-integration` will always skip the tests unless `ENABLE_WEBHOOK` is expiclity set to `"true"`. So, to run these integration tests as a part of the CI, we need to set `ENABLE_WEBHOOK` here - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/addon-operator/openshift-addon-operator-main.yaml#L61

Reply:
What do you think about forcing that env var via the `test-e2e` target?
That way we don't have to touch CI as dependency of this PR.

```
diff --git a/Makefile b/Makefile
index 66cd487..916e008 100644
--- a/Makefile
+++ b/Makefile
@@ -230,6 +230,7 @@ test-integration: config/deploy/deployment.yaml \
 .PHONY: test-integration
 
 # legacy alias for CI/CD
+test-e2e: ENABLE_WEBHOOK=true # force webhook tests in CI/CD
 test-e2e: test-integration
 .PHONY: test-e2e
```

Reply:
What I'm concerned about is if we'll ever be required to _not_ run the webhook tests locally as a part of `test-e2e` - if ever the need arises, can we further override it? I.e, `ENABLE_WEBHOO=false make test-e2e`? IIUC, technically, `test-e2e` is supposed to test everything, and there should be no reason to leave out the webhook tests. 😄 

Either way, I'm not too opinionated about the approach we take. Enforcing it through make also sounds great to me!

Reply:
`test-e2e` is only used by CI right now, because we renamed "e2e" to "integration", soooo it should not impact anyone.
Except people using their shell history :D